core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -51,6 +51,30 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
+  @GET
+  @Path("executors/{executorId}/threads")
+  def threadDump(@PathParam("executorId") executorId: String): Array[ThreadStackTrace] =
+    withUI { ui =>
+    if (executorId != SparkContext.DRIVER_IDENTIFIER && !executorId.forall(Character.isDigit)) {
```

Comment:
This whole block now needs to be indented further, and a `{` added in the declaration.

(Or, if you call the parameter `execId` instead, everything fits in one line...)